src/services/services.ts

```diff
@@ -1293,9 +1293,11 @@ namespace ts {
             // It needs to be cleared to allow all collected snapshots to be released
             hostCache = undefined;
 
-            // Make sure all the nodes in the program are both bound, and have their parent
-            // pointers set property.
-            program.getTypeChecker();
+            if (!host.isLanguageServiceEnabled || host.isLanguageServiceEnabled()) {

```

Comment:
i would say remove this whole section altogether, and instead make sure all calls to `createLanguageServiceSourceFile` pass `setParents` to true.

Reply:
https://github.com/Microsoft/TypeScript/pull/23033